iocBoot/templates/st.cmd.template

```diff
@@ -0,0 +1,103 @@
+#!../../bin/rhel7-x86_64/{{binary_name}}
+
+< envPaths
+epicsEnvSet("IOCNAME", "{{name}}" )
+epicsEnvSet("ENGINEER", "{{user}}" )
+epicsEnvSet("LOCATION", "{{prefix}}" )
+epicsEnvSet("IOCSH_PS1", "$(IOCNAME)> " )
+
+cd "$(TOP)"
+
+# Run common startup commands for linux soft IOC's
+< /reg/d/iocCommon/All/pre_linux.cmd
+
+# Register all support components
+dbLoadDatabase("dbd/{{binary_name}}.dbd")
+{{binary_name}}_registerRecordDeviceDriver(pdbbase)
+
+cd "$(TOP)/db"
+
+epicsEnvSet("ASYN_PORT",     "{{asyn_port}}")
+epicsEnvSet("IPADDR",        "{{plc_ip}}")
+epicsEnvSet("AMSID",         "{{plc_ams_id}}")
+epicsEnvSet("IPPORT",        "{{plc_ads_port}}")
+
+adsAsynPortDriverConfigure("$(ASYN_PORT)","$(IPADDR)","$(AMSID)","$(IPPORT)", 1000, 0, 0, 50, 100, 1000, 0)
+
+{% if symbols.Symbol_FB_MotionStage %}
+epicsEnvSet("MOTOR_PORT",    "{{motor_port}}")
+epicsEnvSet("PREFIX",        "{{prefix}}{{delim}}")
+epicsEnvSet("ECM_NUMAXES",   "{{symbols.Symbol_FB_MotionStage|length}}")
+epicsEnvSet("NUMAXES",       "{{symbols.Symbol_FB_MotionStage|length}}")
+
+EthercatMCCreateController("$(MOTOR_PORT)", "$(ASYN_PORT)", "$(NUMAXES)", "200", "1000")
+
+#define ASYN_TRACE_ERROR     0x0001
+#define ASYN_TRACEIO_DEVICE  0x0002
+#define ASYN_TRACEIO_FILTER  0x0004
+#define ASYN_TRACEIO_DRIVER  0x0008
+#define ASYN_TRACE_FLOW      0x0010
+#define ASYN_TRACE_WARNING   0x0020
+#define ASYN_TRACE_INFO      0x0040
+asynSetTraceMask("$(ASYN_PORT)", -1, 0x41)
+
+#define ASYN_TRACEIO_NODATA 0x0000
+#define ASYN_TRACEIO_ASCII  0x0001
+#define ASYN_TRACEIO_ESCAPE 0x0002
+#define ASYN_TRACEIO_HEX    0x0004
+asynSetTraceIOMask("$(ASYN_PORT)", -1, 2)
+
+#define ASYN_TRACEINFO_TIME 0x0001
+#define ASYN_TRACEINFO_PORT 0x0002
+#define ASYN_TRACEINFO_SOURCE 0x0004
+#define ASYN_TRACEINFO_THREAD 0x0008
+asynSetTraceInfoMask("$(ASYN_PORT)", -1, 5)
+
+#define AMPLIFIER_ON_FLAG_CREATE_AXIS  1
+#define AMPLIFIER_ON_FLAG_WHEN_HOMING  2
+#define AMPLIFIER_ON_FLAG_USING_CNEN   4
+
+{% for motor in symbols.Symbol_FB_MotionStage | sort(attribute='nc_axis.axis_number') %}
```

Comment:
This fails on my project because I don't have any `FB_MotionStage` in my `Main`. I only have my `DUT_MotionStage` and a `FB_PPM` for my common component stuff, with `FB_MotionStage` being embedded underneath `FB_PPM`.

Reply:
I think we'll need to determine what's an acceptable level of nesting, and code around that. The initial assumption was that there will always be `Main.FB_MotionStage`s.

Reply:
I agree, I just wanted to note what happened with my project

Reply:
However, is there any reason we have to target the function block here instead of the stage data type? The stage data type is the thing the IOC needs, so it's the thing you're guaranteed to find.

Reply:
I think either could work, @ZLLentz, and maybe the DUT would be even easier. Changes would have to be made to add a `Symbol_DUT_MotionStage` with similar supporting methods/properties that `Symbol_FB_MotionStage` has.

Reply:
I opened up an issue in pytmc about this - please continue the discussion there: https://github.com/slaclab/pytmc/issues/98